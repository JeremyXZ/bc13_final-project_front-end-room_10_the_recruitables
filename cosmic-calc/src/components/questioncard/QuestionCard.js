import React from "react";
import "./questionCard.css";

export default function QuestionCard(props) {
  return (
    <div className="questionDiv">
      <h2 className="h2QuestionGame">{props.noOfQuestions + ")  "}</h2>
      <h2 className="h2QuestionGame">
        {props.num1} x {props.num2} =
      </h2>
      <input
        className="inputGame"
        type="number"
        value={props.answer}
        onChange={(e) => props.setAnswer(e.target.value)}
        onKeyDown={(e) => {
          // console.log(e);
          if (e.key === "Enter") {
            props.checkAnswer();
            props.setAnswer("");
          }
        }}
      />
    </div>
  );
}
