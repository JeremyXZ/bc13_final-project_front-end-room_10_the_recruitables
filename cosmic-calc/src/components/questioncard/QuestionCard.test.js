import React from "react";
import "@testing-library/jest-dom";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";


import QuestionCard from "./QuestionCard";

describe("<QuestionCard/>", () => {
  const mockedProps = {
    value1: 50,
    value2: 20,
    operation: "+",
    answerInput: 70,
    equals: "=",
    inputType: 'number',    
    checkAnswer: jest.fn(),
    setAnswerInput: jest.fn()
  };

  it("successfully renders the component, matches snapshot", () => {
    const { asFragment } = render(<QuestionCard {...mockedProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the provided 'value1', 'value2', 'operation', 'equals' prop content to the DOM" , () => {
    render(<QuestionCard  {...mockedProps} />);
    expect(screen.getByTestId('questionCard').textContent).toBe("50 + 20 =");
    
  });

  it("Typing in a value and pressing 'Enter' key fires  the provided 'checkAnswer' and 'setAnswerInput' callback props" , () => {
    render(<QuestionCard {...mockedProps} />);
    const input = screen.getByRole('spinbutton')
    userEvent.type(input, `${mockedProps.answerInput}, {enter}`)
    expect(mockedProps.checkAnswer).toHaveBeenCalled();
    expect(mockedProps.setAnswerInput).toHaveBeenCalled();
  
  });

  
  it("Only typing in a value without pressing 'Enter' key fires 'setAnswerInput' but not 'checkAnswer' callback prop." , () => {
  render(<QuestionCard {...mockedProps} />);
  const input = screen.getByRole('spinbutton')
  userEvent.type(input, `${mockedProps.answerInput}`)
  expect(mockedProps.setAnswerInput).toHaveBeenCalled();  
  expect(mockedProps.checkAnswer).not.toHaveBeenCalled();
  
});


it("Typing in the provided 'anwerInput' prop renders 'displayValue' to the input field" , () => {
render(<QuestionCard {...mockedProps} />);
const input = screen.getByRole('spinbutton')
userEvent.type(input, `${mockedProps.answerInput}`)
expect(input).toHaveValue(mockedProps.answerInput)
        
});

it("Typing in a value and pressing 'Enter' key clears the input field" , () => {
  render(<QuestionCard {...mockedProps} />);
  let input = screen.getByRole('spinbutton')
  userEvent.type(input, `${mockedProps.answerInput}, {enter}`)
  input = screen.findByRole('spinbutton')
  expect(input.value).toBeUndefined()
  
});
      
})





   
