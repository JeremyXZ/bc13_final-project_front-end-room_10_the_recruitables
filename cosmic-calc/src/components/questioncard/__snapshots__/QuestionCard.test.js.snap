// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionCard/> successfully renders the component, matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="questionDiv"
  >
    <h2
      class="h2QuestionGame"
    >
      undefined)  
    </h2>
    <h2
      class="h2QuestionGame"
      data-testid="questionCard"
    >
      50 + 20 =
    </h2>
    <input
      class="inputGame"
      type="number"
      value="70"
    />
  </div>
</DocumentFragment>
`;
